src/lib/net_state.rs

```diff
@@ -12,8 +12,10 @@ pub struct NetState {
     pub routes: Vec<Route>,
 }
 
-pub fn get_state() -> Result<NetState, NisporError> {
-    let ifaces = get_ifaces()?;
-    let routes = get_routes(&ifaces)?;
-    Ok(NetState { ifaces, routes })
+impl NetState {
+    pub fn new() -> Result<NetState, NisporError> {
+        let ifaces = get_ifaces()?;
+        let routes = get_routes(&ifaces)?;
+        Ok(NetState { ifaces, routes })
+    }
```

Comment:
This will break API and a lot need to be changed: C and Python binding, test cases.

Please move this patch into separate PR for discussion.

Reply:
Moved to https://github.com/nispor/nispor/pull/27